### PR TITLE
open container for customizations during runtime

### PR DESCRIPTION
**Pull request by fuhbar**
You are able to mount in your customization scripts under
`/scripts/toran-proxy/local/*.sh` which - if existent - will get sourced
by `launch.sh` during container startup.

### CODE REVIEW

File: scripts/launch.sh [hunk 1]
```diff
@@ -10,6 +10,17 @@ source $SCRIPTS_DIRECTORY/install/ssh.sh
 source $SCRIPTS_DIRECTORY/install/php.sh
 source $SCRIPTS_DIRECTORY/install/nginx.sh
```
Comment (cedvan) [inline scripts/launch.sh]: Please create and load a script to load customs scripts to  `$SCRIPTS_DIRECTORY/install/custom-scripts.sh`

```
source $SCRIPTS_DIRECTORY/install/nginx.sh
source $SCRIPTS_DIRECTORY/install/custom-scripts.sh
```

Paste this to `custom-scripts.sh`

```
# If any user provided scripts are found source them
LOCALD="$SCRIPTS_DIRECTORY/local"
FILES="$( [ -e $LOCALD ] && find $LOCALD -type f ! -name '\.*' -a -name '*.sh')"
if [ -n "$FILES" ]; then
  echo "Running local customization scripts ..."
  for file in $FILES; do
    echo "-> $file ..."
    source $file
  done
fi
```

**Comment by cedvan** [comment]
Hi,

Thanks you for contributing :smile: !

**Review by cedvan — CHANGES_REQUESTED** [review]

**Comment by fuhbar** [comment]
I moved this code into a seperate script as you suggested.

**Review by cedvan — APPROVED** [review]

**Comment by fuhbar** [comment]
Do you agree on the location where to find the customization scripts (`/scripts/toran-proxy/local`)? I thought `local` implies site local changes. What do you think?

File: README.md [hunk 2]
```diff
@@ -110,6 +110,27 @@ Below is the complete list of available options that can be used to customize yo
...
+```bash
+export custdir=/tmp/toran-customs
+mkdir -p $custdir
+echo "echo 'hello world'" > $custdir/hello.sh
+docker run --name toran-proxy -d \
+    -p 443:443 \
+    -v $custdir:/scripts/toran-proxy/local
+    cedvan/toran-proxy:1.5.3-1
+```
+
```
Comment (cedvan) [inline README.md]: One white line useless

**Review by cedvan — COMMENTED** [review]
Please update `**latest**` section to `Changelog.md`

**Comment by cedvan** [comment]
Hum, must be `/data/toran-proxy/scripts` :wink:

File: README.md [hunk 3]
```diff
@@ -110,6 +110,27 @@ Below is the complete list of available options that can be used to customize yo
...
+The container `launch.sh` script expects custom scripts to be found under
+`/scripts/toran-proxy/local/*.sh`. These scripts just get source in order. 
+
+```bash
+export custdir=/tmp/toran-customs
+mkdir -p $custdir
+echo "echo 'hello world'" > $custdir/hello.sh
+docker run --name toran-proxy -d \
+    -p 443:443 \
+    -v $custdir:/scripts/toran-proxy/local
```
Comment (cedvan) [inline README.md]: `-v $custdir:/data/toran-proxy/scripts`

File: scripts/install/custom-scripts.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# If any user provided scripts are found source them
+LOCALD="$SCRIPTS_DIRECTORY/local"
```
Comment (cedvan) [inline scripts/install/custom-scripts.sh]: `LOCALD="$DATA_DIRECTORY/scripts"`

**Review by cedvan — CHANGES_REQUESTED** [review]

File: Changelog.md [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 **latest**
+- Add hook directory for user customizations to be mounted into the container
```
Comment (cedvan) [inline Changelog.md]: Add custom scripts directory `/data/toran-proxy/scripts` for user customizations to be executed on startup

File: README.md [hunk 6]
```diff
@@ -110,6 +110,26 @@ Below is the complete list of available options that can be used to customize yo
...
 
+## Add customization scripts
+
+For scenarios where the degree of configurability his image offers via the
+above listed options is not sufficient, you are able to add container local
+customization scripts which will get executed during container runtime. Here
+you can add for example sed calls which further tweak the nginx configuration. 
+
+The container `launch.sh` script expects custom scripts to be found under
+`/scripts/toran-proxy/local/*.sh`. These scripts just get source in order.
```
Comment (cedvan) [inline README.md]: `/scripts/toran-proxy/local/*.sh` => `/data/toran-proxy/scripts/*.sh`

File: Changelog.md [hunk 7]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 **latest**
+- Add custom scripts directory /data/toran-proxy/scripts for user customizations to be executed on startup
```
Comment (cedvan) [inline Changelog.md]: /data/toran-proxy/scripts => \`/data/toran-proxy/scripts\`

**Review by cedvan — CHANGES_REQUESTED** [review]

**Review by cedvan — APPROVED** [review]

**Comment by cedvan** [comment]
Nice, now just rebase and squash your commits, and I'll go merge !

**Comment by fuhbar** [comment]
Here we go

**Comment by cedvan** [comment]
Great ! 

Thank you for contributing :smile:

**Pull request merged**